### DISCUSSION TITLE
run open ssl command: openssl pkcs12 -in elastic-certificates.p12 -nocerts -nodes > client.key.  Terminal request "enter import password", what is import password

### DISCUSSION THREAD

**Discussion by MartinZheng-ary** [openssl/openssl#22849] [Q&A] [answered]
I need to separate a p12 file for kibana .

run：openssl pkcs12 -in elastic-certificates.p12 -nocerts -nodes > client.key

then I need enter import password
![image](https://github.com/openssl/openssl/assets/76100856/2ca0ea51-1eed-401d-9e7b-7366e54e0217)

So I enter the password I used to creat this p12 file

after that get the error in above image.

I tried several and confirmed that the password I used to creat this p12 file is correct

then run： openssl pkcs12 -in elastic-certificates.p12 -out client.key -nocerts -nodes -password pass:wei1205

I got the same result.

![image](https://github.com/openssl/openssl/assets/76100856/f1f67ab9-5a78-4394-ae6d-942833fc3338)

If I enter a password randomly, then I get "Mac verify error: invalid password?"

Someone could help me, THX

**Comment by mattcaswell** [comment]
The error indicates a failure to load the `RC2-40-CBC` encryption algorithm which the input PCKS12 file was encrypted with. This encryption algorithm is considered legacy and insecure.

You might try amending your command line to add the options `-provider default -provider legacy`. This should load the default provider (as normal) as well as the legacy provider, which has support for additional legacy algorithms in it (including RC2-40-CBC).

**Reply by MartinZheng-ary** [reply]
Thank you. But this solution is not work 
Thank you again.

**Reply by MartinZheng-ary** [reply, marked as answer]
According your idea, I fins another post and solve this problem, just add '-leagcy' after p12 file name, like:

openssl pkcs12 -in elastic-certificates.p12 -legacy -nocerts -nodes > client.key

It worked!!! Yeah!!

Thanks! Wish you a good day!
